api/base/serializers.py

```diff
@@ -282,7 +281,15 @@ def lookup_attribute(self, obj, lookup_field):
         bracket_check = _tpl(lookup_field)
         if bracket_check:
             source_attrs = bracket_check.split('.')
-            return get_nested_attributes(obj, source_attrs)
+            # If you are using a nested attribute for lookup, and you get the attribute wrong, you will not get an
+            # error message, you will just not see that field. This allows us to have slightly more dynamic use of
+            # nested attributes in relationship fields.
+            try:
+                return_val = get_nested_attributes(obj, source_attrs)
+            except KeyError:
```

Comment:
Mmmmaybe. I think if we're passing an object in, it's going to have a predictable structure, so it should raise an error. But let's say that we're passing in a pointer node rather than a dictionary called pointer, so the case where `params.pointer.id` wouldn't return a key error. If there's no pointer in that case, the type would be none and it would return none at the top of the `get_nested_attributes`. This problem comes in when `params.pointer` exists but `params.pointer` equals `{}`. It's not none, but it doesn't have the key. But if it is an object rather than a dictionary, then the only reason the Attribute wouldn't exist is if you don't know what you're passing in, right? That seems like something that won't happen dynamically, and handling the error would make it harder to debug. 

I could be thinking about it wrong, but I think we should wait until we cause the error before we try to fix this one, unless you see the flaw in my thinking.